File: EmployeePortal/ClientApp/src/app/models/clinic.ts
import { Address } from './address';
import { Room } from './room';
import { Tax } from './tax';
import { ServiceCategory } from './service-category';

export interface Clinic {
  clinicId: number;
  name: string;
  address?: Address;
  addressId?: number;
  clinicTaxes?: ClinicTax[];
  clinicRooms?: ClinicRoom[];
  companyId?: number;
}

export interface ClinicTax {
  clinicId: number;
  clinic: Clinic;
  taxId: number;
  tax: Tax;
}

export interface ClinicRoom {
  clinicId: number;
  clinic: Clinic;
  roomId: number;
  room: Room;
}

File: EmployeePortal/Models/clinic.cs
using EmployeePortal.Models;
using Newtonsoft.Json;
using System.Collections.Generic;

namespace EmployeePortal.Models
{
    public class Clinic {

        public int clinicId { get; set; }
        public string name { get; set; }

        public int? addressId { get; set;}
        public virtual Address address { get; set; }
        public virtual ICollection<ClinicRoom> clinicRooms { get; set; }
        public virtual ICollection<ClinicTax> clinicTaxes { get; set; }
        public int? companyId { get; set; }
    }

    public class ClinicTax
    {
        public int ClinicId { get; set; }
        [JsonIgnore]
        public Clinic Clinic { get; set; }
        public int TaxId { get; set; }
        public Tax Tax { get; set; }
    }

    public class ClinicRoom
    {
        public int ClinicId { get; set; }
        [JsonIgnore]
        public Clinic Clinic { get; set; }
        public int RoomId { get; set; }
        public Room Room { get; set; }
    }
}